chore(docs): Update gatsby-slice.md to reference styled components non-support (#37525)
gatsbyjs/gatsby: docs/docs/reference/built-in-components/gatsby-slice.md

## Restrictions on using `<Slice>` placeholder

> ### JSS support and Styled component support

Reviewer: ```suggestion
### JSS and Styled Components Support
```

> Using Styled-components or JSS in slices won't work, Alternatively you can use emotion or normal css.

Reviewer: ```suggestion
Using Styled-components or JSS within Slice components is currently [not supported](https://github.com/gatsbyjs/gatsby/issues/37278). Alternatively, you can use emotion or normal CSS.
```
